public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -804,40 +790,52 @@ export class ElasticDatasource
     return lastValueFrom(this.getFields());
   }
 
-  getTagValues(options: any) {
+  getTagValues(options: { key: string }) {
     const range = this.timeSrv.timeRange();
     return lastValueFrom(this.getTerms({ field: options.key }, range));
   }
 
-  targetContainsTemplate(target: any) {
+  targetContainsTemplate(target: ElasticsearchQuery) {
     if (this.templateSrv.containsTemplate(target.query) || this.templateSrv.containsTemplate(target.alias)) {
       return true;
     }
 
-    for (const bucketAgg of target.bucketAggs) {
-      if (this.templateSrv.containsTemplate(bucketAgg.field) || this.objectContainsTemplate(bucketAgg.settings)) {
-        return true;
+    if (target.bucketAggs) {
+      for (const bucketAgg of target.bucketAggs) {
+        if (isBucketAggregationWithField(bucketAgg) && this.templateSrv.containsTemplate(bucketAgg.field)) {
+          return true;
+        }
+        if (this.objectContainsTemplate(bucketAgg.settings)) {
+          return true;
+        }
       }
     }
 
-    for (const metric of target.metrics) {
-      if (
-        this.templateSrv.containsTemplate(metric.field) ||
-        this.objectContainsTemplate(metric.settings) ||
-        this.objectContainsTemplate(metric.meta)
-      ) {
-        return true;
+    if (target.metrics) {
+      for (const metric of target.metrics) {
+        if (!isMetricAggregationWithField(metric)) {
+          continue;
+        }
+        if (metric.field && this.templateSrv.containsTemplate(metric.field)) {
+          return true;
+        }
+        if (metric.settings && this.objectContainsTemplate(metric.settings)) {
+          return true;
+        }
+        if (isMetricAggregationWithMeta(metric) && this.objectContainsTemplate(metric.meta)) {
+          return true;
+        }
       }
     }
 
     return false;
   }
 
-  private isPrimitive(obj: any) {
+  private isPrimitive(obj: unknown) {
     if (obj === null || obj === undefined) {
       return true;
     }
-    if (['string', 'number', 'boolean'].some((type) => type === typeof true)) {
+    if (['string', 'number', 'boolean'].some((type) => type === typeof obj)) {
```

Comment:
It seems that there was a bug here 🤔

Reply:
hmmm.. i recommend being careful with that change, maybe do it in a separate PR.
problem is, if i'm reading this right, in the current (main-branch) version of the code, `isPrimitive` ALWAYS returns true.  (please correct me if i'm wrong).

so, if we change this:
- we will fix the obvious bug in the code
- but, the whole thing (i think this is called from `datasource.targetContainsTemplate`) will now behave differently, maybe causing some other bug.

so we should learn/understand what real effect this will cause :-(.